sdk/python/lib/pulumi/runtime/_callbacks.py

```diff
@@ -89,23 +90,58 @@ async def shutdown(cls):
         for servicer in cls._servicers:
             await servicer._server.wait_for_termination(timeout=0)
 
-    # aio handles this being async but the pyi typings don't expect it.
     async def Invoke(
-        self, request: callback_pb2.CallbackInvokeRequest, context
+        self,
+        request: callback_pb2.CallbackInvokeRequest,
+        context: aio.ServicerContext[
+            callback_pb2.CallbackInvokeRequest, callback_pb2.CallbackInvokeResponse
+        ],
     ) -> callback_pb2.CallbackInvokeResponse:
         log.debug(f"Invoke callback {request.token}")
         callback = self._callbacks.get(request.token)
         if callback is None:
-            context.abort(
+            await context.abort(
```

Comment:
This needs to be awaited. This also never returns, so we don't need to raise the exception below.

Reply:
Can we leave a comment that this never returns

Reply:
In fact maybe just leave the exception with the message changed to say "Unreachable, abort never returns"